<a href="https://colab.research.google.com/github/Julia-BarCar/Estrutura-de-Dados-II/blob/main/Aula_03_EstDin%C3%A2micas.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Testes professora

In [1]:
# Identidade do Objeto
a = 5
b= a
print (int(a))
print (int(b))

5
5


In [3]:
lista1 = [1,2, 3]
lista2 = lista1
lista3 = lista1[:]
print(lista1)
print(lista2)
print(lista3)

[1, 2, 3]
[1, 2, 3]
[1, 2, 3]


#Classe Node

In [ ]:
class Node:
    def __init__(self, dado):
        self.dado = dado
        self.proximo = None

#Atividade 01
Responda:

1. Qual será a saída?
2. a e b representam o mesmo objeto?
3. Por quê?
4. O que aconteceria com b = a.copy()?

In [7]:
a = [10, 20, 30]
b = a

b.append(40)

print("a:", a)
print("b:", b)
print("id(a):", id(a)) # Mostra o id na memória
print("id(b):", id(b))

a: [10, 20, 30, 40]
b: [10, 20, 30, 40]
id(a): 135772668651840
id(b): 135772668651840


##Respostas
1.

a: [10, 20, 30, 40]

b: [10, 20, 30, 40]

id(a): 135772668651840

id(b): 135772668651840

2. Sim
3. Porque a atribuição b = a copia a referência do objeto (a lista), e não o seu conteúdo. Portanto, qualquer mudança feita em b também altera a.
4. b faria uma cópia separada de a para si, fazendo mudanças como append ocorreream apenas em b.

#Atividade 02
Crie:

n1 = Node("A")

n2 = Node("B")

n3 = Node("C")

Conecte:

A → B → C → None

Depois percorra a estrutura e produza:

A
B
C

In [3]:
class Node:
    def __init__(self, dado):
        self.dado = dado
        self.proximo = None

n1 = Node("A")
n2 = Node("B")
n3 = Node("C")

n1.proximo = n2
n2.proximo = n3
n3.proximo = None
print(n1.dado)
print(n1.proximo.dado)
print(n1.proximo.proximo.dado)

A
B
C


#Atividade 03
Perguntas:
1. Qual é o problema?
2. Por que o programa não termina?
3. Qual linha deve ser acrescentada?
4. Qual será a saída depois da correção?

In [1]:
from io import BufferedReader
class Node:
    def __init__(self, dado):
        self.dado = dado
        self.proximo = None

n1 = Node(10)
n2 = Node(20)
n3 = Node(30)

n1.proximo = n2
n2.proximo = n3
n3.proximo = None

atual = n1

while atual is not None:
    # print(atual.valor)
      print(atual.dado)
      atual = atual.proximo

10
20
30


##Respostas
1. Na linha 11, no "print(atual.valor)
2. A classe básica do node não atribuiu o atributo valor
3. Não há necessidade de adicionar uma linha a mais para o erro dito, apenas ver na classe Node qual o nome atribuído, nesse caso é .dado; mas é necessário adicionar uma condição de parada real pra não haver um looping infinito adicionando "atual = atual.proximo"
4. 10
20
30

#Sistema de Clínica

Paciente :
- nome,
- idade,
- níveis de prioridade.

Dinâmica :
- Pacientes chegam de forma contínua e saem quando são atendidos.

Contrato :
- Adicionar paciente,
- remover,
- listar espera,
- checagem de lista vazia